software-engineer/pt_BR/do-and-donts.md

```diff
@@ -0,0 +1,99 @@
+| ✅ | Vista-se confortavelmente. Normalmente você não precisa usar roupas elegantes. |
+| ✅ | Procure se manter calmo, na medida do possível. Entendemos que uma entrevista não é o ambiente mas confortável do mundo, por isso, vamos te apoiar e tentar trazer o ambiente mais "leve" que pudermos. |
+| ⚠️ | Dê preferência por deixar a webcam ligada. Caso isso não seja possível ou seja desconfortável para você, não tem problema, é só falar com a pessoa entrevistadora sobre isso. |
+
+### Introdução
+
+| | Coisas |
+| --- | --- |
+| ✅ | Apresente-se brevemente. |
+| ✅ | Mencione pontos interessantes que sejam relevantes para a função para a qual você está se candidatando. |
+| ✅ | Demonstre entusiasmo e você naturalmente parecerá mais envolvente. |
+| ❌ | Passe muito tempo se apresentando. Quanto mais tempo você gasta falando, menos tempo você tem para codificar. |
+
+### Ao receber a pergunta
+
+| | Coisas |
+| --- | --- |
+| ✅ | Repita a pergunta para quem estiver te entrevistando, para confirmar que o entendimento está correto. |
+| ✅ | Esclareça quaisquer suposições que pareça inconscientemente. Muitas questões são subespecificadas propositalmente. Por exemplo, se perguntado como você implementaria um sistema de upload, questione possíveis "pontas soltas" em uma povável solução para um sistema de upload, como, por exemplo, quais formatos de arquivos serão permitidos. |
+| ✅ | Esclareça o formato e o intervalo de entrada. Pergunte se a entrada pode ser considerada bem formada e não nula. |
+| ✅ | Trabalhe com um pequeno exemplo para garantir que você entendeu a pergunta. |
+| ✅ | Explique uma abordagem de alto nível, mesmo que seja de força bruta. |
+| ✅ | Melhore a abordagem e otimize. Reduza o trabalho duplicado e armazene em cache cálculos repetidos. |
+| ✅ | Se estiver com dificuldades em prosseguir, pense nos problemas relacionados que você viu antes e como eles foram resolvidos. |
+| ❌ | Ignore as informações fornecidas a você. Cada peça é importante. |
+| ❌ | Passe para a codificação imediatamente. |
+| ❌ | Comece a codificar sem a luz verde da pessoa entrevistadora. |
+| ❌ | Demonstre muita insegurança sobre sua abordagem ou análise. |
+
+### Codifique sua solução
+
+| | Coisas |
+| --- | --- |
+| ✅ | Explique o que você está codificando / digitando para a pessoa entrevistadora. Deixe claro o que você está tentando alcançar. |
+| ✅ | Pratique um bom estilo de codificação. Nomes de variáveis ​​claros, espaçamento de operador consistente, recuo adequado, etc.
+| ✅ | Digite / escreva a uma velocidade razoável. Muito lento não é bom. |
+| ✅ | Tanto quanto possível, escreva código compilável real, não pseudocódigo. |
+| ✅ | Escreva de forma modular. Extraia pedaços de código repetido em funções. |
+| ✅ | Peça permissão para usar funções triviais sem ter que implementá-las; você economiza algum tempo. |
+| ✅ | Use as dicas da pessoa entrevistadora. |
+| ✅ | Demonstre domínio da linguagem de programação escolhida. |
```

Comment:
Pra mim essa dica ficou um pouco etérea e pouco prática. Também vejo um risco aqui, de isso fazer as pessoas começarem a trazer soluções maiores do que o necessário, só para mostrar domínio da linguagem.
Faz sentido? 🤔

Reply:
Concordo com esse ponto. Acho que isso inclusive vai na contra-mão do que falamos no início da entrevista de coding, que é uma dinâmica de pair programming normal e que a pessoa deve ficar totalmente à vontade para fazer buscas no google, por exemplo... 
Acho que já cobrimos o que poderia ser dito por esse ponto quando dissemos lá em cima para a pessoa escolher a linguagem de programação que tenha mais familiaridade, voto por remover esse ponto...

Reply:
Removed!